src/Database/LSMTree/Internal/BlobRef.hs

```diff
@@ -67,8 +69,33 @@ blobRefSpanSize = fromIntegral . blobSpanSize . rawBlobRefSpan
 --
 -- See 'Database.LSMTree.Common.BlobRef' for more info.
 --
-newtype WeakBlobRef m h = WeakBlobRef (RawBlobRef m h)
-  deriving newtype (Show, NFData)
+data WeakBlobRef m h = WeakBlobRef {
+      weakBlobRefFile :: {-# NOUNPACK #-} !(BlobFile m h)
+    , weakBlobRefSpan :: {-# UNPACK #-}   !BlobSpan
+    }
+  deriving stock (Show)
+
+-- | A \"strong\" reference to a blob within a blob file. The blob file remains
+-- open while the strong reference is live. Thus it is safe to do I\/O to
+-- retrieve the blob based on the reference. Strong references must be released
+-- using 'releaseBlobRef' when no longer in use (e.g. after completing I\/O).
+--
+data StrongBlobRef m h = StrongBlobRef {
+      strongBlobRefFile :: {-# NOUNPACK #-} !(BlobFile m h)
+    , strongBlobRefSpan :: {-# UNPACK #-}   !BlobSpan
+    }
+  deriving stock (Show)
```

Comment:
Okay, I now see that we do make strong blob references explicit as a type. It might make more sense in the commit history of this PR to introduce strong blob references earlier? It's probably not important enough to change, but it confused me while going through the PR patch by patch

Reply:
I guess I could have introduced it earlier, but not used it. In the end it turns out that strong references are never used outside of the `BlobRef` module. They're intermediate things used during I/O for blob retrieval.